src/main/java/stringCalculator/BasicOperations.java

```diff
@@ -0,0 +1,35 @@
+package stringCalculator;
+
+import java.util.Arrays;
+import java.util.function.BiFunction;
+
+public enum BasicOperations {
+
+	PLUS("+", (num1, num2) -> num1 + num2),
+	SUBTRACT("-", (num1, num2) -> num1 - num2),
+	MULTIPLY("*", (num1, num2) -> num1 * num2),
+	DIVIDE("/", (num1, num2) -> num1 / num2)
+	;
+
+	private final String operator;
+	private final BiFunction<Integer, Integer, Integer> expression;
+
+	BasicOperations(
+		String operator,
+		BiFunction<Integer, Integer, Integer> expression
+	) {
+		this.operator = operator;
+		this.expression = expression;
+	}
+
+	public static BasicOperations match(String operator) {
+		return Arrays.stream(BasicOperations.values())
+			.filter(bo -> bo.operator.equals(operator))
```

Comment:
```suggestion
			.filter(bo -> bo.equalsOperator(operator))
```

enum이라 그럴 일은 없겠지만 한 줄에 점이 2개 이상 들어가면 NPE 위험이 있습니다.
데이터를 꺼내와서 연산하는거라 객체지향 구현에도 조금 애로사항이 생기구요.
이 도메인 클래스의 역할이면 이 클래스에 선언해주시는게 좋습니다.
디미터 법칙이라는 이론이 있으니 읽어보세요.

https://tecoble.techcourse.co.kr/post/2020-06-02-law-of-demeter/